resource/reapi/bindings/go/src/pkg/types/types.go

```diff
@@ -0,0 +1,55 @@
+package types
+
+/*****************************************************************************\
+ * Copyright 2024 Lawrence Livermore National Security, LLC
+ * (c.f. AUTHORS, NOTICE.LLNS, LICENSE)
+ *
+ * This file is part of the Flux resource manager framework.
+ * For details, see https://github.com/flux-framework.
+ *
+ * SPDX-License-Identifier: LGPL-3.0
+\*****************************************************************************/
+
+/*
+#include "reapi_cli.h"
+
+typedef enum {
+	MATCH_UNKNOWN = 0
+	MATCH_ALLOCATE = 1
+	MATCH_ALLOCATE_ORELSE_RESERVE = 2
+	MATCH_ALLOCATE_W_SATISFIABILITY = 3
+	MATCH_SATISFIABILITY = 4
+} match_op_enum;
```

Comment:
Do we need to declare the enum here? Or can we just use the one included with reapi_cli.h?

Reply:
We don't need to - you can remove this. It's only there for the time being when the actual reapi_cli.h is not added via cgo (was for prototyping). You'll need to keep it for the tests to pass before you do that, but please nuke away when you add the actual binding! :boom: